### PR TITLE
Spell check

### LINKED ISSUE #362: Initial Spell check
To catch some minor typos etc. 

* [ ] https://github.com/nunit/docs/pull/339#discussion_r439733375

### PR DESCRIPTION
**Pull request by SeanKilleen**
Going over things with cSpell and setting up the dictionary as I go.

Resolves #362.

Prepares for #388.

I:

* ran `npm install -g cSpell`
* ran `cSpell --config ./cSpell.json "docs/**/*.md"` over and over

## Remaining Work

* [x] Populate "words" list to remove spelling issues
* [x] Break into "words" and "ignore" lists (words shows up in suggestions; ignore does not.)
* [x] Capture regex to help ignore terms within markdown links, e.g. xrefs

### CODE REVIEW

**Comment by SeanKilleen** [comment]
As of now, 734 issues found via cSpell.

**Comment by SeanKilleen** [comment]
Down to #447.

**Comment by SeanKilleen** [comment]
Down to 372.

**Comment by SeanKilleen** [comment]
Down to 203.

**Comment by SeanKilleen** [comment]
Down to 99.

**Comment by SeanKilleen** [comment]
Down to zero spelling errors found! 🎉 

Next up: separating actual words from ignored terms

File: docs/articles/developer-info/Issue-Tracking.md [hunk 1]
```diff
@@ -72,9 +72,9 @@ The goal with these is to keep team members from having to do tedious rescanning
 
 Labels starting with **closed:** indicate the status of the bug at closing and should only appear on closed bugs. Please remember to apply one of these when closing a bug as it makes it easier to review the list of closed bugs without opening each one to see what the disposition was.
 
-* **closed:done** The work called for is done, i.e. the bug is fixed or the feature/enhancement is implemented.
+* **`closed:done** The work called for is done, i.e. the bug is fixed or the feature/enhancement is implemented.
```
Comment (mikkelbu) [inline docs/articles/developer-info/Issue-Tracking.md:75]: Is the ` on purpose? Same comment for line 77

Comment (SeanKilleen) [reply docs/articles/developer-info/Issue-Tracking.md:75]: Nope! Definitely a mistake. Saw your comment when I came here to review for those sorts of issues. 👍 thanks! Will fix.

Comment (SeanKilleen) [inline docs/articles/developer-info/Issue-Tracking.md:75]:
```suggestion
* **closed:done** The work called for is done, i.e. the bug is fixed or the feature/enhancement is implemented.
```

File: docs/articles/developer-info/Issue-Tracking.md [hunk 2]
```diff
@@ -72,9 +72,9 @@ The goal with these is to keep team members from having to do tedious rescanning
...
 * **closed:duplicate** The issue is a duplicate of one that we are working. A comment should indicate the issue number.
-* **closed:notabug** The issue (generally a bug) is not valid or the feature already exists. There should be an explanatory comment.
+* **closed:notabug`** The issue (generally a bug) is not valid or the feature already exists. There should be an explanatory comment.
```
Comment (SeanKilleen) [inline docs/articles/developer-info/Issue-Tracking.md:77]:
```suggestion
* **closed:notabug** The issue (generally a bug) is not valid or the feature already exists. There should be an explanatory comment.
```

File: cSpell.json [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    "activeconfig",
+    "netstandard",
+    "Gallio",
+    "Codice",
+    "mytest",
+    "Rebracer",
+    "myassembly"
+  ],
+  "ignoreRegExpList": [
+    "\\((.*)\\)", // Markdown links
```
Comment (SeanKilleen) [inline cSpell.json:122]: 📓 I know that comments in JSON aren't correct but this tooling allows us to get away with it and I figured the extra context was worth it.

File: docs/articles/developer-info/Packaging-the-V2-Adapter.md [hunk 4]
```diff
@@ -15,12 +15,12 @@ The version numbers follow the basic principles of [semantic versioning].
 
 The version numbers have to be edited in the following files, and should match:
 
-* **AssemblyInfo.cs**,  found in the NUnitTestAdapter project
--- change both file and assembly version number
-* **source.extensions.vsixmanifest**, found under the NUnitTestAdapterInstall project
--- change Version tag
-* **nunit-vs-adapter.build**, found under the Solution Items folder. -- change the version number, but only use the two first digits.
-* **license.rtf**, found under the NUNitTestAdapterInstall project.  If the major/minor number has changed, update that here, 2nd line. If year is changed, update copyright years accordingly.
+| File | Change |
```
Comment (SeanKilleen) [inline docs/articles/developer-info/Packaging-the-V2-Adapter.md:18]: 📓 Updated the formatting to a table while I was at it.

File: docs/articles/developer-info/Packaging-the-V3-Adapter.md [hunk 5]
```diff
@@ -31,13 +31,13 @@ The version numbers follow the basic principles of [semantic versioning].
...
 The version numbers have to be edited in the following files, and should match:
 
-* **AssemblyInfo.cs**,  found in the NUnitTestAdapter project
-   -- change both file and assembly version number
-* **source.extensions.vsixmanifest**, found under the NUnitTestAdapterInstall project
-   -- change Version tag
-* **appveyor.yml**, found under the Solution Items folder. -- change the version number, but only use the three first digits.
-* **build.cake**, found under the Solution Items folder. -- change the version number, but only use the three first digits.
-* **license.rtf**, found under the NUnit3TestAdapterInstall project.  If the copyright year has changed, update accordingly.
+| File | Change |
```
Comment (SeanKilleen) [inline docs/articles/developer-info/Packaging-the-V3-Adapter.md:34]: 📓 Updated the formatting to a table while I was at it.

File: docs/articles/nunit/getting-started/breaking-changes.md [hunk 6]
```diff
@@ -58,7 +58,7 @@ Other breaking changes are grouped in the following tables.
 | SuiteAttribute               | No longer supported. |
 | System.MTAThreadAttribute    | No longer treated as `RequiresMTAAttribute`             |
 | System.STAThreadAttribute    | No longer treated as `RequiresSTAAttribute`             |
-| TearDown and OneTimeTearDown | There is a change to the logic by which teardown methods are called. See [Setup and Teardown Changes](xref:setupteardownchanges for details. |
+| TearDown and OneTimeTearDown | There is a change to the logic by which teardown methods are called. See [Setup and Teardown Changes](xref:setupteardownchanges) for details. |
```
Comment (SeanKilleen) [inline docs/articles/nunit/getting-started/breaking-changes.md:61]: 📓 Fixed this broken link while I was at it.

**Pull request merged**